### PR TITLE
CHANGE: Make DualSenseHIDInputReport public so it can be used in child devices

### PR DESCRIPTION
**Pull request by lewish-unity**
### Description

Made the input report for the DualsenseHID gamepad public so that it can be used in child devices as input reports. 
This fixes [ISXB-652](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-652) in combination with fixes in other packages.

### Changes made

- Made DualSenseHIDInputReport public
- Added tests for recognising the Dualsense Controller 

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [x] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [x] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=1945) <br/>All committers have signed the CLA.

**Review by ekcoh — CHANGES_REQUESTED** [review]
Changes LGTM apart from CHANGELOG.md not being updated as part of this PR. Could you add a line under our unreleased version under CHANGED that DualSenseHIDInputReport was changed from internal to public visibility? 

(We should make sure to release a minor revision next time due to this change.)

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,9 @@ however, it has to be formatted properly to pass verification tests.
   set by `InputSystem.settings.maxEventsBytesPerUpdate`. This additional information is available in development builds
   only.
 
+### Changed
+- Changed DualSenseHIDInputReport from internal to public visibility
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Minor, but when referencing type/code use `Code Formatting` since an .md file

**Review by ekcoh — APPROVED** [review]

**Review by ritamerkl — APPROVED** [review]

**Comment by Pauliusd01** [comment]
Don't see anything wrong with the changes but this does not fix [ISXB-652](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-652), I still see the actual result from the bug description

image

**Review by Pauliusd01 — APPROVED** [review]

**Review by timkeo — APPROVED** [review]

**Comment by ekcoh** [comment]
@lewish-unity I noticed there is a CI failure on this one https://unity-ci.cds.internal.unity3d.com/job/38452361?utm_source=github I believe you need to update version number in package.json from "1.8.3" to "1.9.0" for API change to not generate an error.

**Comment by lewish-unity** [comment]
Will do, I thought this might happen with a different PR but will update it in my branch

**Pull request merged**